src/lib/cli/rules/common.ts

```diff
@@ -41,6 +97,166 @@ export type NewRule = {
     events: string;
     /** If the new rule is core */
     isRecommended: boolean;
-    /**  Usage categories that the new rule applies to */
+    /** Usage categories that the new rule applies to */
     useCase?: UseCase;
+    /** Path prefix for dependencies */
+    prefix: string;
+    /** If the rule works with local files */
+    worksWithLocalFiles: Boolean;
+    /** If a rule is external */
+    external: Boolean;
+}
+
+export enum RuleType {
+    core = 'core',
+    external = 'external',
+    externalRule = 'externalRule'
+}
+
+/** List rule categories. */
+const categories = [];
+
+for (const enumValue in Category) {
+    if (Category.hasOwnProperty(enumValue)) {
+        categories.push({ name: Category[enumValue] });
+    }
+}
+
+/** List of different use cases of a rule. */
+const useCases = [
+    {
+        name: 'DOM',
+        value: 'dom'
+    },
+    {
+        name: 'Resource Request',
+        value: 'request'
+    },
+    {
+        name: 'Third Party Service',
+        value: 'thirdPartyService'
+    },
+    {
+        name: 'JS injection',
+        value: 'jsInjection'
+    }
+];
+
+/** List of questions to prompt the user. */
+export const questions = (type: RuleType) => {
+    return [{
+        message: `Is this a package with multiple rules? (yes)`,
+        name: 'multi',
+        type: 'confirm',
+        when: () => {
+            return type === RuleType.external;
+        }
+    },
+    {
+        default(answers) {
+            return answers.multi ? 'newPackage' : 'newRule';
+        },
+        message(answers) {
+            return `What's the name of this new ${answers.multi ? 'package' : 'rule'}?`;
+        },
+        name: 'name',
+        type: 'input'
+    },
+    {
+        default(answers) {
+            return `new awesome ${answers.name}`;
+        },
+        message(answers) {
+            return `What's the description of this new ${answers.multi ? 'package' : 'rule'} '${answers.name}'?`;
+        },
+        name: 'description',
+        type: 'input'
+    },
+    {
+        default() {
+            return type === RuleType.core;
+        },
+        message: 'Is it a recommended rule',
+        name: 'recommended',
+        type: 'confirm',
+        when(answers) {
+            return !answers.multi;
+        }
+    },
+    {
```

Comment:
Can we not make this question for third party rules? Maybe we should create another category  `other` that we assign directly to those rules? The category is not relevant to third party rules and is probably going to confuse the users.

Reply:
why not? For example for `axe` the category is `accessibility`

Reply:
I should have said external rules, not third party. Sorry about that.

Reply:
No problem. Done!